crates/accelerate/src/unitary_synthesis.rs

```diff
@@ -0,0 +1,1101 @@
+    match target_basis_list {
+        Ok(basis_list) => {
+            EULER_BASES
+                .iter()
+                .enumerate()
+                .filter_map(|(idx, gates)| {
+                    if !gates.iter().all(|gate| basis_list.contains(gate)) {
+                        return None;
+                    }
+                    let basis = EULER_BASIS_NAMES[idx];
+                    Some(basis)
+                })
+                .for_each(|basis| target_basis_set.add_basis(basis));
+        }
+        Err(_) => target_basis_set.support_all(),
+    }
+    if target_basis_set.basis_supported(EulerBasis::U3)
+        && target_basis_set.basis_supported(EulerBasis::U321)
+    {
+        target_basis_set.remove(EulerBasis::U3);
+    }
+    if target_basis_set.basis_supported(EulerBasis::ZSX)
+        && target_basis_set.basis_supported(EulerBasis::ZSXX)
+    {
+        target_basis_set.remove(EulerBasis::ZSX);
+    }
+    target_basis_set
+}
+
+fn apply_synth_dag(
+    py: Python<'_>,
+    out_dag: &mut DAGCircuit,
+    out_qargs: &[Qubit],
+    synth_dag: &DAGCircuit,
+) -> PyResult<()> {
+    for out_node in synth_dag.topological_op_nodes()? {
+        let  NodeType::Operation(mut out_packed_instr) = synth_dag.dag()[out_node].clone() else {
+            panic!("DAG node must be an instruction")
+        };
+        let synth_qargs = synth_dag.get_qargs(out_packed_instr.qubits);
+        let mapped_qargs: Vec<Qubit> = synth_qargs
+            .iter()
+            .map(|qarg| out_qargs[qarg.0 as usize])
+            .collect();
+        out_packed_instr.qubits = out_dag.qargs_interner.insert(&mapped_qargs);
+        out_dag.push_back(py, out_packed_instr)?;
+    }
+    out_dag.add_global_phase(py, &synth_dag.get_global_phase())?;
+    Ok(())
+}
+
+fn apply_synth_sequence(
+    py: Python<'_>,
+    out_dag: &mut DAGCircuit,
+    out_qargs: &[Qubit],
+    sequence: &TwoQubitUnitarySequence,
+) -> PyResult<()> {
+    let mut instructions = Vec::new();
+    for (gate, params, qubit_ids) in sequence.gate_sequence.gates() {
+        let gate_node = match gate {
+            None => sequence.decomp_gate.operation.standard_gate(),
+            Some(gate) => *gate,
+        };
+        let mapped_qargs: Vec<Qubit> = qubit_ids.iter().map(|id| out_qargs[*id as usize]).collect();
+        let new_params: Option<Box<SmallVec<[Param; 3]>>> = match gate {
+            Some(_) => Some(Box::new(params.iter().map(|p| Param::Float(*p)).collect())),
+            None => Some(Box::new(sequence.decomp_gate.params.clone())),
+        };
+        let instruction = PackedInstruction {
+            op: PackedOperation::from_standard(gate_node),
+            qubits: out_dag.qargs_interner.insert(&mapped_qargs),
+            clbits: out_dag.cargs_interner.get_default(),
+            params: new_params,
+            extra_attrs: ExtraInstructionAttributes::new(None, None, None, None),
+            #[cfg(feature = "cache_pygates")]
```

Comment:
Here's something that might look a bit less complex.
```suggestion
            extra_attrs: ExtraInstructionAttributes::default()
```
Unless you have a good reason for doing the manual assignment.